### PR TITLE
Fix zip serialization for file > 2GiB for Windows

### PR DESCRIPTION
**Pull request by peterjc123**
`long long == int64_t != long` in MSVC

### CODE REVIEW

**Comment by ezyang** [comment]
It should be safe to just unconditionally use LongLong here, right? 64-bit will always be big enough to store.

**Comment by peterjc123** [comment]
> It should be safe to just unconditionally use LongLong here, right? 64-bit will always be big enough to store.

Actually int64_t = long long with MSVC

File: torch/csrc/jit/python/init.cpp [hunk 1]
```diff
@@ -759,7 +759,11 @@ void initJITBindings(PyObject* module) {
         auto res =
             PyObject_CallMethod(buffer_.ptr(), "readinto", "O", memview.get());
         if (res) {
-          int64_t i = PyInt_AsLong(res);
+#ifdef _MSC_VER
+          int64_t i = PyLong_AsLongLong(res);
+#else
+          int64_t i = PyLong_AsLong(res);
```
Comment (ezyang) [inline torch/csrc/jit/python/init.cpp:765]: So specifically, I'm suggesting eliminating this if-branch of the ifdef and just always use the MSC_VER version.

Comment (peterjc123) [reply torch/csrc/jit/python/init.cpp:765]: Then it will lead to a static casting here for Unix. Is that okay?

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**